src/server/serve.rs

```diff
@@ -145,7 +141,7 @@ where
                 let matches = router.visit(request.uri().path());
                 let next = router.resolve(&matches, request.params_mut());
 
-                async {
+                Box::pin(async {
```

Comment:
A bit of scope creep here but probably beneficial if there is an additional layer of indirection here. This is the boundary where library code stops and app code begins.

Reply:
This means that the majority of the library code will be work with data on the stack. App code, will have a boxed future for every middleware function that is called + 1. I'm assuming that a `Pin<Box<Future>>` is required for the work stealing thread pool in tokio to poll a future created from a separate task. I'm also assuming that this is what we want in that, the connection will be polled and bytes will be written to the socket from the task spawned when the connection is accepted while app code can be polled from any task (worker thread). Curious if anyone on the tokio team has thoughts or opinions here.